mapper-processor/src/main/java/com/datastax/oss/driver/internal/mapper/processor/entity/DefaultEntityFactory.java

```diff
@@ -111,31 +123,57 @@ public EntityDefinition getDefinition(TypeElement processedClass) {
         }
 
         String getMethodName = getMethod.getSimpleName().toString();
-        boolean regularGetterName = getMethodName.startsWith("get");
-        boolean booleanGetterName =
-            getMethodName.startsWith("is")
-                && (typeMirror.getKind() == TypeKind.BOOLEAN
-                    || context.getClassUtils().isSame(typeMirror, Boolean.class));
-        if (!regularGetterName && !booleanGetterName) {
+
+        // Skip a few false positives when we don't require a "get" prefix:
+        if (getterStyle == GetterStyle.SHORT
+            && !mutable
+            && (getMethodName.equals("toString") || getMethodName.equals("hashCode"))) {
+          continue;
+        }
+        if (isScalaCaseClass
+            && (getMethodName.equals("productPrefix")
+                || getMethodName.equals("productArity")
+                || getMethodName.equals("productIterator")
+                || getMethodName.startsWith("copy$default$"))) {
```

Comment:
don't we also need `canEqual` and `productElement`? (https://www.scala-lang.org/api/2.12.4/scala/Product.html)